src/check_release.rs

```diff
@@ -350,19 +373,57 @@ pub(super) fn run_check_release(
                 ),
                 Color::Ansi(AnsiColor::Red),
                 true,
-            )
-            .expect("print failed");
+            )?;
+        } else if produced_warnings {
+            writeln!(config.stderr())?;
+            config.shell_print(
+                "Summary",
+                "no semver update required",
+                Color::Ansi(AnsiColor::Green),
+                true,
+            )?;
+        } else {
+            unreachable!();
```

Comment:
If there's a good description of what appears to have gone wrong for us to hit this, it would be great to put it here.

Even if right now it may seem obvious to you and me, we're still 200+ lines into a very complex function, and we might not think it's so obvious in 6 months.